File: wpf_cs/BBUtil.cs
using System;
using System.Windows.Input;

namespace BBUtil{
	//キーボードの状態を管理するやつ
	static class CursorKey{
		static bool _left,_up,_right,_down;
		public static bool left{get{return _left;}}
		public static bool up{get{return _up;}}
		public static bool right{get{return _right;}}
		public static bool down{get{return _down;}}

		public static Action<KeyEventArgs> keyDown=keyState(true);
		public static Action<KeyEventArgs> keyUp=keyState(false);
		static Action<KeyEventArgs> keyState(bool state){
			return new Action<KeyEventArgs>(e=>{
				switch(e.Key){
					case Key.Left: _left=state; break;
					case Key.Up: _up=state; break;
					case Key.Right: _right=state; break;
					case Key.Down: _down=state; break;
				}
			});
		}
	}
}

File: wpf_vb/BBUtil.vb
Option Strict On

Imports System
Imports System.Windows.Input

NameSpace BBUtil
	'キーボードの状態を管理するやつ
	Module CursorKey
		Private _left,_up,_right,_down As Boolean
		ReadOnly Property left As Boolean
			Get
				Return _left: End Get: End Property
		ReadOnly Property up As Boolean
			Get
				Return _up: End Get: End Property
		ReadOnly Property right As Boolean
			Get
				Return _right: End Get: End Property
		ReadOnly Property down As Boolean
			Get
				Return _down: End Get: End Property

		Public  keyDown As Action(Of KeyEventArgs)=keyState(true)
		Public  keyUp As Action(Of KeyEventArgs)=keyState(false)
		Function keyState(state As Boolean) As Action(Of KeyEventArgs)
			Return Sub(e)
				Select Case e.Key
					case Key.Up: _up=state
					case Key.Left: _left=state
					case Key.Right: _right=state
					case Key.Down: _down=state
				End Select
			End Sub
		End Function
	End Module
End NameSpace
